framework/src/interfaces/Coupleable.C

```diff
@@ -321,6 +322,27 @@ Coupleable::getDefaultVectorValue(const std::string & var_name)
   if (default_value_it == _default_vector_value.end())
   {
     VectorVariableValue * value = new VectorVariableValue(_coupleable_max_qps, 0);
+    bool already_warned = false;
+    for (unsigned int qp = 0; qp < _coupleable_max_qps; ++qp)
+      for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+      {
+        try
+        {
+          (*value)[qp](i) = _c_parameters.defaultCoupledValue(var_name, i);
+        }
+        catch (const std::out_of_range &)
+        {
+          if (!already_warned)
+            mooseWarning(
+                "You supplied less than 3 arguments for the default vector value for variable ",
+                var_name,
+                ". Did you accidently leave something off? We are going to assign 0s, assuming "
+                "this "
+                "was intentional.");
+          already_warned = true;
```

Comment:
Use mooseDoOnce macro instead - way easier.

Reply:
I chose not to use `mooseDoOnce` in case (in the admittedly unlikely event) I wanted to warn for multiple variables